File: NeetCode/Array/containsDuplicate.java
import java.util.Hashset;
import java.util.Scanner;
import java.util.set;

class containsDuplicate 
{
    public boolean hasDuplicate(int []nums)
    {
        Set<integer> hashset = new Hashset<>();

        for (int num: nums)
        {
            if (hashset.contains(num))
            {
                return true;
            }
            hashset.add(num);
        }
        return false;
    }

    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter numbers separated by space: ");

        String[] input = scanner.nextLine().split(" ");

        int[] nums = new int[input.Length];

        for (int i = 0; i < nums.Length; i++)
        {
            nums[i] = Integer.parseInt(input[i]);
        }

        Solutions s = new containsDuplicate();

        boolean result = s.hasDuplicate(nums);

        System.out.println(result);

        scanner.close();
    }
}

File: NeetCode/Array/contains_duplicate.py
from typing import List

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        
        hashset = set()

        for num in nums:
            if num in hashset:
                return True
            hashset.add(num)
        
        return False
    
if __name__ == "__main__":

    nums = [x for x in input().split()]

    s = Solution()
    print(s.hasDuplicate(nums))